src/Views/dialogView.cpp

```diff
@@ -0,0 +1,242 @@
+	_accepted = false;
+	_status = COMMON_DIALOG_STATUS_RUNNING;
+	_btnTouched = _btnFocus = -1;
+	if (_result) {
+		_result->status = COMMON_DIALOG_STATUS_RUNNING;
+	}
+	_type = type;
+}
+
+int DialogView::GetGlowCycleAlpha() {
+	float alphaCount;
+	if (_focusGlowCycle <= DIALOG_FOCUS_GLOW_HALF_CYCLE_FRAMES) {
+		alphaCount = _focusGlowCycle;
+	} else {
+		alphaCount = DIALOG_FOCUS_GLOW_HALF_CYCLE_FRAMES * 2 - _focusGlowCycle;
+	}
+	return int(alphaCount / DIALOG_FOCUS_GLOW_HALF_CYCLE_FRAMES * 255);
+}
+
+void DialogView::HandleBtnFocus(const Input& input) {
+	auto btnFocusBefore = _btnFocus;
+	if (input.KeyNewPressed(SCE_CTRL_RIGHT)) {
+		switch (_type) {
+			case DIALOG_TYPE_OK:
+				_btnFocus = 0;
+				break;
+			case DIALOG_TYPE_YESNO:
+				_btnFocus = 1;
+				break;
+		}
+		log_printf(DBG_DEBUG, "SCE_CTRL_RIGHT, _btnFocus %i", _btnFocus);
+	}
+	if (input.KeyNewPressed(SCE_CTRL_LEFT) || input.KeyNewPressed(SCE_CTRL_UP) || input.KeyNewPressed(SCE_CTRL_DOWN)) {
+		_btnFocus = 0;
+		log_printf(DBG_DEBUG, "SCE_CTRL_LEFT||SCE_CTRL_UP||SCE_CTRL_DOWN , _btnFocus %i", _btnFocus);
+	}
+	if (input.TouchPressed()) {
+		_btnFocus = -1;
+	}
+	if (_btnFocus >= 0) {
+		if (btnFocusBefore < 0 ||  // focus button after none was focused
+		    (btnFocusBefore != _btnFocus && GetGlowCycleAlpha() < 200)) {  // change focused button and highlight was nearly gone
+			log_printf(DBG_DEBUG, "Resetting glow cycle");
+			_focusGlowCycle = 0;
+		}
+	}
+	if (input.KeyNewPressed(SCE_CTRL_CROSS) && _btnFocus >= 0) {
+		log_printf(DBG_DEBUG, "SCE_CTRL_CROSS, choosing button %i", _btnFocus);
+		_status = COMMON_DIALOG_STATUS_FINISHED;
+		if (_type == DIALOG_TYPE_YESNO && _btnFocus == 1) {
+			_accepted = true;
+		}
+	}
+}
+
+int DialogView::GetTouchedBtnIdx(const Input &input) {
+	if (!input.TouchPressed()) return -1;
+	switch (_type) {
+		case DIALOG_TYPE_OK: {
+			auto btn_tl = Point(DIALOG_BTN_1_OF_1_INNER_LEFT_X, DIALOG_BTN_INNER_TOP_Y);
+			auto btn_br = Point(DIALOG_BTN_1_OF_1_INNER_RIGHT_X, DIALOG_BTN_INNER_BOT_Y);
+			if (input.TouchInRectangle(Rectangle(btn_tl, btn_br))) {
+				return 0;
+			}
+			break;
+		}
+		case DIALOG_TYPE_YESNO:
+			// NO button
+			auto btn_tl = Point(DIALOG_BTN_1_OF_2_INNER_LEFT_X, DIALOG_BTN_INNER_TOP_Y);
+			auto btn_br = Point(DIALOG_BTN_1_OF_2_INNER_RIGHT_X, DIALOG_BTN_INNER_BOT_Y);
+			if (input.TouchInRectangle(Rectangle(btn_tl, btn_br))) {
+				return 0;
+			}
+			// YES button
+			btn_tl = Point(DIALOG_BTN_2_OF_2_INNER_LEFT_X, DIALOG_BTN_INNER_TOP_Y);
+			btn_br = Point(DIALOG_BTN_2_OF_2_INNER_RIGHT_X, DIALOG_BTN_INNER_BOT_Y);
+			if (input.TouchInRectangle(Rectangle(btn_tl, btn_br))) {
+				return 1;
+			}
+			break;
+	}
+	return -1;
+}
+
+void DialogView::HandleBtnTouch(const Input& input) {
+	if (input.TouchPressed()) {
+		if (input.TouchNewPressed()) {
+			_btnTouched = GetTouchedBtnIdx(input);
+		} else if (_btnTouched >= 0){
+			if (_btnTouched != GetTouchedBtnIdx(input)) {
+				_btnTouched = -1;
+			}
+		}
+	} else {
```

Comment:
` else if`

Reply:
I think that `else if` would decrease the readability of the code…